doc/integrator/ngeo.rst.tmpl

```diff
@@ -140,112 +175,124 @@ Creating your own component
 
-Then register your component to this module:
+.. code:: typescript
 
-.. code:: javascript
+  // File ...static-ngeo/js/custom/helloWorld.ts;
+  import {html, TemplateResult, CSSResult, css} from 'lit';
+  import {customElement, property} from 'lit/decorators.js';
+  import GmfBaseElement from 'gmfapi/elements/BaseElement';
+  import i18next from 'i18next';
 
-   class aController {}
-   module.controller('<projectController>', aController);
-   const aDirective = function() {};
-   module.directive('<projectDirectivename>', aDirective);
+  @customElement('hello-world')
+  export default class GmfAuthForm extends GmfBaseElement {
+    @property({type: String}) name = 'Unknow';
 
-And export this module:
+    connectedCallback(): void {
+      super.connectedCallback();
+    }
 
-.. code:: javascript
+    static styles: CSSResult[] = [
+      ...GmfBaseElement.styles,
+      css`
+        .name {
+          color: green;
+        }
+      `,
+    ];
 
-   export default module;
+    protected render(): TemplateResult {
+      return html`
+        <div>
+          <span>${i18next.t('Hello')}&nbsp;</span>
+          <span class="name">${this.name}</span>
+        </div>
+      `;
+    }
+  }
 
-Finally add your module in your main controller app file:
+Then load your component in the controller of one of your interfaces:
 
 .. code:: javascript
 
-   import myCustomComponent from '../myComponent/Component.js';
-   // Update the already declared module at the end of the controller with:
-   const module = angular.module('<AppInterface>', [
-     ...
-     myCustomComponent.name,
-   ];
+   import GmfAuthForm from '../custom/helloWorld';
 
 At this point, if in your <interface> HTML app file you declare your component, it should be loaded:
 
 .. code:: html
 
-   <project-directivename></project-directivename>
+   <hello-world name="Demo"></hello-world>
 
-If you have no error and no result, verify that the module is correctly declared and loaded. AngularJS can't
-throw errors for a component that it doesn't know and having a custom markup in the HTML is not an error
-for the HTML.
+It should print ``Hello Demo`` (with "Demo" in a green color).
 
-.. note::
+If you have no error and no result, verify that the component is correctly declared and loaded.
+A custom markup in the HTML is not an error and no error will be throw if the component is
+not loaded at all.
 
-   To load a partial (HTML) file for a component you can use this syntax inside the declaration of your
-   component: ``template: require('./component.html')`` (from ``./Component.js`` file).
+Every information should be stored in a ``store``. Our store are based on the
+``rxjs`` library. Existing store in ngeo are in `srcapi/store`.
+
+Logic between a store and a component should be stored in a singleton ("service").
+singleton exported instance of a class. Example in ngeo: `src/auth/service.ts`.
+
+If possible, keep as many logic out of your component, in standard classes. It will
```

Comment:
Where should the component logic be, if not in the component?

Reply:
```suggestion
If possible, keep as many logic out of your component in standard classes. For instance in reusable external classes, static functions, "helpers" and "utils" files etc. It will
```

Reply:
Thanks I understand
Suggestion for the two sentences: 
"We recommend to minimize the amount of code within your component, and to use reusable external classes, static functions, helper and utils files instead."

Reply:
Updated